graph/src/components/ethereum/adapter.rs

```diff
@@ -942,16 +944,27 @@ pub fn blocks_with_triggers(
         ));
     }
 
-    let logger1 = logger.clone();
+    let logger1 = logger.cheap_clone();
+    let logger2 = logger.cheap_clone();
+    let eth_clone = eth.cheap_clone();
     Box::new(
         trigger_futs
             .concat2()
-            .join(adapter.clone().block_hash_by_block_number(
-                &logger,
-                chain_store.clone(),
-                to,
-                true,
-            ))
+            .join(
+                adapter
+                    .clone()
+                    .block_hash_by_block_number(&logger, chain_store.clone(), to, true)
+                    .then(move |to_hash| match to_hash {
+                        Ok(n) => n.ok_or_else(|| {
+                            warn!(logger2,
+                                    "Ethereum endpoint is behind";
+                                    "url" => eth_clone.url()
```

Comment:
Can we only log the hostname here? URLs may include `username:password@host/path` and we don't want that to show up in logs.

Reply:
Yep, we just need enough to figure out which provider's dashboard to look at